internal/tezos/get_receipt.go

```diff
@@ -100,20 +100,22 @@ func (c *tezosConnector) TransactionReceipt(ctx context.Context, req *ffcapi.Tra
 					script, err = c.client.GetContractScript(ctx, tx.Destination)
 					if err != nil {
 						log.L(ctx).Error("error getting contract script: ", err)
+						return nil, "", err
 					}
 				}
 				if len(tx.Result().Errors) > 0 {
-					errorMessage := ""
+					var errs error
 					for _, err := range tx.Result().Errors {
-						errorMessage += err.Error()
+						errs = errors.Join(errs, err)
 					}
-					extraInfo.ErrorMessage = &errorMessage
+					return nil, "", err
 				}
 				if prim := tx.Metadata.Result.Storage; prim != nil {
 					val := micheline.NewValue(script.StorageType(), *prim)
 					m, err := val.Map()
 					if err != nil {
 						log.L(ctx).Error("error parsing contract storage: ", err)
+						return nil, "", err
```

Comment:
Even if we get the smart contract error - the transaction is still considered successful and will return an error as a part of the transaction's receipt